### DISCUSSION TITLE
Consider adding a TOML parser

### DISCUSSION THREAD

**Discussion by MicahElliott** [babashka/babashka#1725] [Ideas]
**Is your feature request related to a problem? Please describe.**
I would like to parse `.toml` files and don't see a way presently with any babashka included libs. I'd expect this to be a common/useful scripting use case since [toml](https://toml.io/en/v1.0.0) is widely used.

**Describe the solution you'd like**
The [tonsky/toml-clj](https://github.com/tonsky/toml-clj) lib seems to be working well for it with jvm-clojure, and has no deps (though I could be wrong about that, and I do see [some java files](https://github.com/tonsky/toml-clj/tree/main/src-java/me/tonsky/toml), so **wondering if it could be added to bb.**

**Describe alternatives you've considered**
I couldn't figure out how to use `toml-clj` since it's not already included in babashka. If there's some way that I could, I'm happy to try. I consulted the [new libraries](https://github.com/babashka/babashka#including-new-libraries-or-classes) section but still wasn't obvious how I could.

Another method I considered would be to shell-out to [yq](https://github.com/mikefarah/yq) to convert input toml file to yaml, and then use bb's clj-yaml.

The other toml parser I see as a candidate is [clj-toml](https://github.com/lantiga/clj-toml). It does have an instaparse dep though, and is rather old and it seems to be not working well with clj v1.12 (maybe just needs an update to newer instaparse).

**Additional context**
I might be able to follow what was done in another PR to add another similar library (clj-yaml?), but first wanted to make sure that you agree that clj-toml is worth adding.

**Comment by borkdude** [comment]
Converting this to a Discussion since it's something that needs more thought and interaction from the community.
